### ISSUE TITLE
V8.1.1253: Core Dump on missing parameter argument

### ISSUE THREAD

**Issue by rakus** [vim/vim#4334]
Vim 8.1.1253 crashes with sigsegv when specific command line parameter are missing a required argument.

Reproducible with:
- `vim --servername`
- `vim --display`

Both parameter are handled in `early_arg_scan(...)`.

OS: Linux x86_64

**Comment by dpelle** [comment]
@brammool  closed this issue, but it's still crashing using the latest vim-8.1.1295:
```
$ ./vim --servername
VIM - Vi IMproved 8.1 (2018 May 18, compiled May  7 2019 23:44:38)
Argument missing after: "--servername"
More info with: "vim -h"
Segmentation fault (core dumped)

$ ./vim --display
VIM - Vi IMproved 8.1 (2018 May 18, compiled May  7 2019 23:44:38)
Argument missing after: "--display"
More info with: "vim -h"
Segmentation fault (core dumped)

$ ./vim --socketid
VIM - Vi IMproved 8.1 (2018 May 18, compiled May  7 2019 23:44:38)
Argument missing after: "--socketid"
More info with: "vim -h"
Segmentation fault (core dumped)
```

When fixed, we can uncomment out added tests for this in `Test_invalid_args()`.
I don't have permission to re-open this issue.

**Comment by brammool** [comment]
Dominique wrote:

> @brammool  closed this issue, but it's still crashing using the latest vim-8.1.1295:
> ```
> $ ./vim --servername
> VIM - Vi IMproved 8.1 (2018 May 18, compiled May  7 2019 23:44:38)
> Argument missing after: "--servername"
> More info with: "vim -h"
> Segmentation fault (core dumped)
> 
> $ ./vim --display
> VIM - Vi IMproved 8.1 (2018 May 18, compiled May  7 2019 23:44:38)
> Argument missing after: "--display"
> More info with: "vim -h"
> Segmentation fault (core dumped)
> 
> $ ./vim --socketid
> VIM - Vi IMproved 8.1 (2018 May 18, compiled May  7 2019 23:44:38)
> Argument missing after: "--socketid"
> More info with: "vim -h"
> Segmentation fault (core dumped)
> ```
> 
> When fixed, we can uncomment out added tests for this in
> `Test_invalid_args()`.  I don't have permission to re-open this issue.

Somehow while doing debugging I reverted the fix.  I'll send it out now.

-- 
Lawmakers made it obligatory for everybody to take at least one bath
each week -- on Saturday night.
		[real standing law in Vermont, United States of America]

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by rakus** [comment]
Fix confirmed with Vim 8.1.1296.  Thank you.
